pyxem/generators/indexation_generator.py

```diff
@@ -98,49 +90,24 @@ def correlate(self,
         """
         signal = self.signal
         library = self.library
-        inplane_rotations = np.deg2rad(inplane_rotations)
-        num_inplane_rotations = inplane_rotations.shape[0]
-        sig_shape = signal.axes_manager.signal_shape
-        signal_half_width = sig_shape[0] / 2
 
         if mask is None:
             # Index at all real space pixels
             mask = 1
 
-        # Create a copy of the library, cropping and padding the peaks to match
-        # max_peaks. Also create rotated pixel coordinates according to
-        # inplane_rotations
-        rotation_matrices_2d = np.array([[[np.cos(t), np.sin(t)], [-np.sin(t), np.cos(t)]] for t in inplane_rotations])
-        cropped_library = {}
-
-        for phase_name, phase_entry in library.items():
-            num_orientations = len(phase_entry['orientations'])
-            intensities_jagged = phase_entry['intensities']
-            intensities = np.zeros((num_orientations, max_peaks))
-            pixel_coords_jagged = phase_entry['pixel_coords']
-            pixel_coords = np.zeros((num_inplane_rotations, num_orientations, max_peaks, 2))
-            for i in range(num_orientations):
-                num_peaks = min(pixel_coords_jagged[i].shape[0], max_peaks)
-                highest_intensity_indices = np.argpartition(intensities_jagged[i], -num_peaks)[-num_peaks:]
-                intensities[i, :num_peaks] = intensities_jagged[i][highest_intensity_indices]
-                # Get and compute pixel coordinates for all rotations about the
-                # center, clipped to the detector size and rounded to integer positions.
-                pixel_coords[:, i, :num_peaks] = np.clip(
-                    (signal_half_width + rotation_matrices_2d @ (
-                        pixel_coords_jagged[i][highest_intensity_indices].T - signal_half_width)).transpose(0, 2, 1),
-                    a_min=0,
-                    a_max=np.array(sig_shape) - 1)
-
-            np.rint(pixel_coords, out=pixel_coords)
-            cropped_library[phase_name] = {
-                'orientations': phase_entry['orientations'],
-                'pixel_coords': pixel_coords.astype('int'),
-                'intensities': intensities,
-                'pattern_norms': np.linalg.norm(intensities, axis=1),
-            }
-
-        matches = signal.map(correlate_library,
-                             library=cropped_library,
+        #TODO: Add extra methods
+        no_extra_methods_yet = True
+        if no_extra_methods_yet:
+            #adds a normalisation to library
+            for phase in library.keys():
```

Comment:
Could you explain a bit what this normalisation is doing please?

Reply:
For (partially) normalised cross correlation (our only method right now) you need to have normalised patterns. You could do this at the comparison stage, but it means you do the same operation at every probe position, which is wasteful. Instead we append the normalisation constant to the library. I'm pretty sure that this code has been like this before.